recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,143 @@
+        # if self.options.with_cjson:
+        #     self._cmake.definitions["cJSON_SEARCH_PATH"] = self.deps_cpp_info["cjson"].rootpath.replace("\\", "/")
+        #     self._cmake.definitions["cJSON_ROOT_DIR"] = self.deps_cpp_info["cjson"].rootpath.replace("\\", "/")
+
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def _patch(self):
+        if self.settings.os == "Windows":
+            if self.options.with_tls:
+                tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                    "${OPENSSL_LIBRARIES}",
+                                    "${OPENSSL_LIBRARIES} crypt32")
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                    "${OPENSSL_LIBRARIES}",
+                                    "${OPENSSL_LIBRARIES} crypt32")
+                # This is so inconsequent, there is ws2_32 meanwhile in the build, but forgotten here
+                tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_passwd" ,"CMakeLists.txt"),
+                                    "${OPENSSL_LIBRARIES}",
+                                    "${OPENSSL_LIBRARIES} ws2_32 crypt32")
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                "install(TARGETS libmosquitto RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")",
+                                "install(TARGETS libmosquitto RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\" ARCHIVE DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")")
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "cpp", "CMakeLists.txt"),
+                                "install(TARGETS mosquittopp RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")",
+                                "install(TARGETS mosquittopp RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\" ARCHIVE DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")")
+
+
+    def build(self):
+        self._patch()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("edl-v10", src=self._source_subfolder, dst="licenses")
+        self.copy("edl-v20", src=self._source_subfolder, dst="licenses")
+        self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib","pkgconfig"))
+        if not self.options.shared:
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.so*")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dll*")
```

Comment:
Are the dll's installed in the `lib` folder?
Or, should it be:
```suggestion
            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.dll*")
```
Should the import library of the dll also be removed?

Reply:
no, that is correct, 
the CMAKE_INSTALL_LIBDIR /ARCHDIR  is explicit set in the cmake file and the files are in lib

Reply:
Oh, that's non-standard.
That also means that `self.cpp_info.bindirs` in `package_info` needs `lib` appended.

But that would be weird when deploying this recipe: dll's in both `bin` and `lib`.
Imho it would be easier to end the `package` method with a glob move of all dll's from `lib` to `bin`.

Reply:
I don't know what you mean , checked curl, openssl, some others, they all put their libs into lib on windows

Reply:
.dll's are binaries, they go in bin
.lib's, .a's are static libraries, they go in lib